images/access-setup/content/bin/setup_compute.sh

```diff
@@ -147,6 +147,33 @@ init() {
   manifests_dir="$WORK_DIR/credentials/manifests"
   mkdir -p "$credentials_dir"
   mkdir -p "$manifests_dir"
+
+  detect_container_engine
+}
+
+detect_container_engine() {
+    CONTAINER_ENGINE="${CONTAINER_ENGINE:-}"
+    ALLOW_ROOTLESS="${ALLOW_ROOTLESS:-false}"
+    if [[ -n "${CONTAINER_ENGINE}" ]]; then
+      return
+    fi
+    CONTAINER_ENGINE="sudo podman"
```

Comment:
Do we need to have root privileges by default? Especially because Tekton Chains was run with just podman run (line 176) without the need for sudo.

Reply:
1. I think apart from local/kcp, we are not using sudo anywhere else. (the local script has a need for sudo)
2. I see, but with the base images we are using such as UBI, we haven't come across this issue.

My point basically is that we should avoid using sudo unless we really need to.

Reply:
1. The same logic is used in `ci/images/quay-upload/image-upload.sh`, `developer/images/devenv/run.sh`, `developer/local/utils.sh`, `images/access-setup/content/bin/setup_compute.sh`.
2. This is not an issue with the base images. This is an issue with systemd in RHEL CSB 8.x that breaks podman in some mysterious ways. Podman never works after a reboot. Sometimes it can be fixed with a `sysctl` command, sometimes it can't be fixed until I reboot the machine.
 
I've added documentation around that bit, please close the conversation if that matches your expectations.

Reply:
Thanks @Mo3m3n. It is very nicely summarized!
I would emphasize on the last line you wrote - "needing to use sudo maybe be an exception in certain environments but it should not be the normal way."

Reply:
@bnallapeta 

1. `podman` is now the default instead of `sudo podman`.
2. Remove references to ALLOW_ROOTLESS (except in `developer/local/kind/setup.sh` where I do have to set it to keep the existing behavior).
3. Updated the doc.

Please resolve if it matches your expectations.

Reply:
@Roming22 Thank you. I have one comment which I have published. But the PR looks good to me.